google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java

```diff
@@ -123,6 +128,14 @@ public void disconnect() {
     }
   }
 
+  @Nullable
+  private File resolveDeploymentSource(DeploymentTask<AppEngineDeploymentConfiguration> task) {
+    DeploymentSource source = task.getSource();
+
+    return task.getConfiguration().isUserSpecifiedArtifact() ?
```

Comment:
agh. I think I made things more difficult by starting to use Path in our code. Will have to think about that.